### ISSUE TITLE
Fast RCNN model output using EvalLibrary

### ISSUE THREAD

**Issue by Vedaevolution** [Microsoft/CNTK#1116]
Dear community,

i'd have question concering the output of the fast rcnn model using c#. Using the standard example i get 4000 output floats. What are those floats? Shouldn't the last layer be softmax? The 4000 output floats are not normalized, so it shouldn't be the output of the softmax layer. Is softmax just used for error evalution and not an actual part of the network. I must admit i'm a little bit confused. 

A little hint would be nice.

Thank you very much,
Stephen

**Comment by cha-zhang** [comment]
Which example are you running? Are these 4000 output floats bounding boxes of the detection?

**Comment by Vedaevolution** [comment]
Thank you for you response. I'm using this example:

[(https://github.com/Microsoft/CNTK/tree/master/Examples/Image/Detection/FastRCNN)]

I found out what those 4000 values are. My configuration is the following:
- i'm using the example above
- i have 2000 input rois over two classes on my own training data
- the 4000 outputvalues from the EvalLibrary are the 2000 rois * 2 classes network outputs WITHOUT softmax applied, which i should have realised is also visible in the brainscript file.

Sry i asked but at first it simply wasn't that obvious to me. Furthermore it would be quite nice to have an evaluation example for c# for rcnn. I finally managed to get it running in pure c# though i did not fully implement selective search in c# and use another roi generation method.

I understand python is an easy to use, easy to learn and state of art scientific language but what got me intrested in CNTK was the fact, that you can use your network models in c# and through this in deployable programms which are not python based. 

It's just meant as an as a suggestion to put in more examples which make it easier for developers to use deep learning aside from scientific research. I really appreciate the CNTK framework and find it a good and helpful tool.

**Comment by mahilleb-msft** [comment]
Thanks for your feedback. Closing, as the original issue seems solved.
